### PR TITLE
OEP-8: copy fixes and formatting

### PR DESCRIPTION
**Pull request by kremalicious**
- copy editing and typo fixes
- all attributes mentioned in text formatted as code
- various example fixes, make all code examples valid json, consistent indentation
- reorganize intro texts

### CODE REVIEW

File: 8/v0.4/README.md [hunk 1]
```diff
@@ -251,46 +223,50 @@ Attribute       |   Type           |   Required    | Description
...
-        "Other"
-      ],
-      "links": [],
-      "tags": [],
-      "updateFrequency": null,
-      "structuredMarkup": []
+{  
+  "service": [
+    {  
+      "serviceDefinitionId": "0",
```
Comment (aaitor) [inline 8/v0.4/README.md]: it should be `index`

File: 8/v0.4/README.md [hunk 2]
```diff
@@ -303,206 +279,224 @@ An asset of type `algorithm` has the following attributes:
...
-        "language": "scala",
-        "format" : "jar",
-        "version": "0.1",
-        "entrypoint" : "ocean-entrypoint.sh",
-        "requirements": [
-            {
+{  
+  "service":[  
+    {  
+      "serviceDefinitionId": "0",
```
Comment (aaitor) [inline 8/v0.4/README.md]: `index` too

File: 8/v0.4/README.md [hunk 3]
```diff
@@ -303,206 +279,224 @@ An asset of type `algorithm` has the following attributes:
...
-                "container": {
-                  "image": "tensorflow/tensorflow",
-                  "tag": "latest",
-                  "checksum": "sha256:cb57ecfa6ebbefd8ffc7f75c0f00e57a7fa739578a429b6f72a0df19315deadc"
-                }
-              },
-              "input": [{
+  "service": [  
+    {  
+      "serviceDefinitionId": "0",
```
Comment (aaitor) [inline 8/v0.4/README.md]: `index` too

File: 8/v0.4/README.md [hunk 4]
```diff
@@ -303,206 +279,224 @@ An asset of type `algorithm` has the following attributes:
...
-          "endpoints": [{
-            "index": 0,
-            "url": "https://my.service.inet:8080/api/v1/weather",
-            "method": "POST",
-            "contentTypes": ["application/json"]          
-          }]
+{  
+  "service": [  
+    {  
+      "serviceDefinitionId": "0",
```
Comment (aaitor) [inline 8/v0.4/README.md]: `index` too

File: 8/v0.4/README.md (hunk 1 above)
Comment (kremalicious) [reply 8/v0.4/README.md]: And should the value be a string or integer? Cause all other `index` values are always integer

Comment (aaitor) [reply 8/v0.4/README.md]: we are using string to make it compatible with the previous `serviceDefinitionId` field and for supporting references

Comment (kremalicious) [reply 8/v0.4/README.md]: ok, cool, just to be sure :-) Pushed the `serviceDefinitionId` -> `index` change

**Comment by aaitor** [comment]
yep, let me know when index is a string and I will approve quickly

**Comment by kremalicious** [comment]
that particular `index` was a string in here already, just renamed the `serviceDefinitionId` key, but left value intact

**Comment by aaitor** [comment]
doing `index` attribute integer for consistency

**Pull request merged**